bootcompiler/src/main/scala/org/mozartoz/bootcompiler/ast/Statements.scala

```diff
@@ -116,11 +117,13 @@ case class NoElseStatement()(val pos: Pos) extends Statement with NoElseCommon {
  *  end
  *  }}}
  */
-case class ForStatement(from: Expression, to: Expression, proc: ProcExpression)(val pos: Pos) extends Statement {
+case class ForStatement(from: Expression, to: Expression, proc: Expression)(val pos: Pos) extends Statement {
+  def procExpr = WrapperUtil.getInnerExpr(proc).asInstanceOf[ProcExpression]
+  
   def syntax(indent: String) = {
-    val variable = proc.args(0)
+    val variable = procExpr.args(0)
     val decl = "for " + variable.syntax(indent) + " in " + from.syntax(indent) + ".." + to.syntax(indent) + " do"
-    decl + "\n" + proc.body.syntax(indent + "   ") + "\n" + indent + "end"
+    decl + "\n" + procExpr.body.syntax(indent + "   ") + "\n" + indent + "end"
```

Comment:
Ah so only ForStatement#syntax depends on it being a ProcExpression :smiley: 
Since this is the only usage, I would do the unwrapping directly in the syntax method:
```scala
val procExpr = WrapperUtil.getInnerExpr(proc).asInstanceOf[ProcExpression]
```